### PR TITLE
115 - branding for homepage, search page, dataset page and resource page

### LINKED ISSUE #6: Create a Fjelltopp theme extension

### PR DESCRIPTION
**Pull request by A-Souhei**
## Description

Closes [https://github.com/fjelltopp/ckanext-fjelltopp-theme/issues/6](https://github.com/fjelltopp/ckanext-fjelltopp-theme/issues/6)

Creates a default theme for Fjelltopp projects. 

I have worked with:

- responsiveness
- fonts styling
- a few templates
- added user management and flow

Basic SASS but can be improved over time.

image
image
image
image
image
image

## Checklist

Put an `x` in the boxes that apply to this pull request (you can also fill these out after opening the pull request).
You may not need to check all boxes.

- [x] The Jira ticket for this issue has been updated to "Ready to Review" or equivalent.
- [x] I have developed these changes in discussion with the appropriate project manager.
- [x] My code follows the general Fjelltopp documentation (see Confluence).
- [ ] I have made corresponding changes to the Fjelltopp documentation (see Confluence).
- [ ] I have rebased this branch with master.
- [ ] New dependency changes have been committed.
- [ ] I have added automated tests that prove my fix is effective or that my feature works.
- [ ] New and existing tests pass locally with my changes.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own code.
- [x] I have assigned at least one reviewer.
- [ ] I have assigned at least one label to this PR: "patch", "minor", "major".

### CODE REVIEW

**Comment by A-Souhei** [comment]
@ChasNelson1990 I am fixing the height of the insights, I will ask for review again later.

File: ckanext/fjelltopp_theme/assets/css/_partials/_fjelltopp.scss [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+.promoted-home .promoted-container {
+    position: relative;
+    min-height: 500px;
+    z-index: 1;
+}
+
+#background-video {
+    position: absolute;
+    right: 0;
```
Comment (ChasNelson1990) [inline ckanext/fjelltopp_theme/assets/css/_partials/_fjelltopp.scss]: I think there is something wrong with your responsive design - this is how the home page looks on my screen.

image

I think this is the offending line.

Comment (ChasNelson1990) [reply ckanext/fjelltopp_theme/assets/css/_partials/_fjelltopp.scss]: Oh, and I think it's meant to go behind the title, right? It doesn't for me - tested on latest Chromium and Google Chrome.

File: ckanext/fjelltopp_theme/assets/css/fjelltopp-theme.css
Comment (ChasNelson1990) [inline ckanext/fjelltopp_theme/assets/css/fjelltopp-theme.css:1]: Do we actually want to commit the css files?
Either we should be compiling scss at run time ourselves (like we do with JS) or using WebAssets to do it for us, right?

File: ckanext/fjelltopp_theme/assets/css/fjelltopp-theme.css.map
Comment (ChasNelson1990) [inline ckanext/fjelltopp_theme/assets/css/fjelltopp-theme.css.map:1]: Same question here - should we be committing map files?

File: ckanext/fjelltopp_theme/templates/home/snippets/promoted.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{% block promote %}
+<div class="promoted promoted-home col-xs-12 col-sm-12 col-lg-12" >
+    {% block backgroundvideo %}
+    <video autoplay muted loop id="background-video">
+    <source src="/images/fjelltopp-bgs/fjelltopp.mp4" type="video/mp4">
```
Comment (ChasNelson1990) [inline ckanext/fjelltopp_theme/templates/home/snippets/promoted.html]: This MP4 file is 70% of the pages size and on slow 3G takes 2 minutes to transfer on slow 3G. Can we convert this to an animated .webp or something?

File: ckanext/fjelltopp_theme/public/fonts/Inter.var.woff2
Comment (ChasNelson1990) [inline ckanext/fjelltopp_theme/public/fonts/Inter.var.woff2:1]: Do we need allll these font files? E.g. what are these variants for?

File: ckanext/fjelltopp_theme/public/fonts/SourceSans3VF-Italic.otf.woff2
Comment (ChasNelson1990) [inline ckanext/fjelltopp_theme/public/fonts/SourceSans3VF-Italic.otf.woff2:1]: Do we need Inter and SourceSans? Is SourceSans just for Zambia?

**Review by ChasNelson1990 — CHANGES_REQUESTED** [review]
There's a lot to review here and it feels like it's all good work but could do with a bit of tidying up and maybe a bit of engineering though about the following:

1. How best to handle compilation
2. How best to handle overriding variables for specific projects, e.g. colours, fonts
3. Compression and format of images
4. Fonts - varieties and formats
5. What belongs in this repo and what might belong elsewhere, e.g. the helper function `get_datahub_stats` - I understand why it feels logical to include it, but this isn't a function about theming and think it probably belongs in it's own mini-extension or something? (Maybe we leave it for now but create an issue for it, who knows...)

File: ckanext/fjelltopp_theme/assets/css/fjelltopp-theme.scss [hunk 3]
```diff
@@ -0,0 +1,1484 @@
...
+@use '_partials/userprofile' as *;
+
+$square-insights-width: math.div(100%, 3);
+$half-window-width: math.div(100%, 2);
+
+
+/* global */
+body {
+    background: config.$light;
+    font-family: config.$default-font-family, serif;
```
Comment (ChasNelson1990) [inline ckanext/fjelltopp_theme/assets/css/fjelltopp-theme.scss]: This should almost certainly default to `sans-serif`

File: ckanext/fjelltopp_theme/public/fonts/Inter.var.woff2
Comment (A-Souhei) [reply ckanext/fjelltopp_theme/public/fonts/Inter.var.woff2:1]: I have removed SourceSans, the font variants actually should not be removed as used by the CSS, eg :  
```
@font-face {
  font-family: 'Inter';
  font-style:  italic;
  font-weight: 100;
  font-display: swap;
  src: url("/fonts/Inter-ThinItalic.woff2?v=3.19") format("woff2"),
       url("/fonts/Inter-ThinItalic.woff?v=3.19") format("woff");
}
```

File: ckanext/fjelltopp_theme/assets/css/_partials/_fjelltopp.scss (hunk 1 above)
Comment (A-Souhei) [reply ckanext/fjelltopp_theme/assets/css/_partials/_fjelltopp.scss]: I have made another scss file that is in charge of loading this background.

File: ckanext/fjelltopp_theme/templates/home/snippets/promoted.html (hunk 2 above)
Comment (A-Souhei) [reply ckanext/fjelltopp_theme/templates/home/snippets/promoted.html]: Now using an animated webp but with a lower quality than original.

**Comment by A-Souhei** [comment]
> There's a lot to review here and it feels like it's all good work but could do with a bit of tidying up and maybe a bit of engineering though about the following:
> 
> 1. How best to handle compilation
> 2. How best to handle overriding variables for specific projects, e.g. colours, fonts
> 3. Compression and format of images
> 4. Fonts - varieties and formats
> 5. What belongs in this repo and what might belong elsewhere, e.g. the helper function `get_datahub_stats` - I understand why it feels logical to include it, but this isn't a function about theming and think it probably belongs in it's own mini-extension or something? (Maybe we leave it for now but create an issue for it, who knows...)

@ChasNelson1990  Should we create another extension for all our utility functions?

**Comment by ChasNelson1990** [comment]
> > There's a lot to review here and it feels like it's all good work but could do with a bit of tidying up and maybe a bit of engineering though about the following:
> > 
> > 1. How best to handle compilation
> > 2. How best to handle overriding variables for specific projects, e.g. colours, fonts
> > 3. Compression and format of images
> > 4. Fonts - varieties and formats
> > 5. What belongs in this repo and what might belong elsewhere, e.g. the helper function `get_datahub_stats` - I understand why it feels logical to include it, but this isn't a function about theming and think it probably belongs in it's own mini-extension or something? (Maybe we leave it for now but create an issue for it, who knows...)
> 
> @ChasNelson1990 Should we create another extension for all our utility functions?

Are talking about `get_datahub_stats` specifically or are there others you are thinking of? I am not a fan of chucking unrelated stuff together in some sort of "general" repo. I think, if it's just `get_datahub_stats` that we're discussing then I would leave it in this repo for now... but if we start collecting multiple helper functions we should think if they start grouping into sensible extensions that add one set of related features to CKAN.

File: package.json [hunk 4]
```diff
@@ -3,6 +3,8 @@
     "sass": "^1.77.8"
   },
   "scripts": {
-    "build": "sass ckanext/fjelltopp_theme/assets/css/fjelltopp-theme.scss ckanext/fjelltopp_theme/assets/css/fjelltopp-theme.css"
+    "compile": "npx sass ckanext/fjelltopp_theme/assets/css/:ckanext/fjelltopp_theme/assets/css/",
+    "watch": "sass --watch ckanext/fjelltopp_theme/assets/css/:ckanext/fjelltopp_theme/assets/css/"
```
Comment (ChasNelson1990) [inline package.json]: Why does `compile` use `npx` but `watch` doesn't?

**Review by ChasNelson1990 — CHANGES_REQUESTED** [review]
I think the hero animation is now tooo low quality - we can probably go for better quality and framerate. Other than that - good to go.

**Comment by A-Souhei** [comment]
> I think the hero animation is now tooo low quality - we can probably go for better quality and framerate. Other than that - good to go.

I will lower the quality by 40%.

**Comment by A-Souhei** [comment]
> I think the hero animation is now tooo low quality - we can probably go for better quality and framerate. Other than that - good to go.

@ChasNelson1990 this is done.

**Review by ChasNelson1990 — DISMISSED** [review]
LGTM

**Review by ChasNelson1990 — APPROVED** [review]
LGTM

**Pull request merged**